google-cloud-examples/src/test/java/com/google/cloud/examples/bigtable/ITHelloWorld.java

```diff
@@ -0,0 +1,162 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.examples.bigtable;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.models.InstanceName;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.AfterClass;
+import org.junit.AssumptionViolatedException;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ITHelloWorld {
+
+  private static final String INSTANCE_PROPERTY_NAME = "bigtable.instance";
+  private static final String TABLE_PREFIX = "table";
+  private static final String COLUMN_FAMILY = "cf1";
+  private static String tableId;
+  private static BigtableDataClient dataClient;
+  private static BigtableTableAdminClient adminClient;
+  private static HelloWorld helloWorld;
+  private static InstanceName instanceName;
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    String targetInstance = System.getProperty(INSTANCE_PROPERTY_NAME);
+    if (targetInstance == null) {
+      dataClient = null;
+      adminClient = null;
+      return;
+    }
+    instanceName = InstanceName.parse(targetInstance);
+    BigtableDataSettings settings =
+        BigtableDataSettings.newBuilder().setInstanceName(instanceName).build();
+    dataClient = BigtableDataClient.create(settings);
+    BigtableTableAdminSettings adminSettings =
+        BigtableTableAdminSettings.newBuilder()
+            .setInstanceName(com.google.bigtable.admin.v2.InstanceName.parse(targetInstance))
+            .build();
+    adminClient = BigtableTableAdminClient.create(adminSettings);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    adminClient.deleteTable(tableId);
+    garbageCollect();
+    dataClient.close();
+    adminClient.close();
+  }
+
+  @Before
+  public void setup() throws IOException {
+    if (adminClient == null || dataClient == null) {
+      throw new AssumptionViolatedException(
+          INSTANCE_PROPERTY_NAME + " property is not set, skipping integration tests.");
+    }
+    tableId = generateTableId();
+    helloWorld = new HelloWorld(instanceName.getProject(), instanceName.getInstance(), tableId);
+    if (!adminClient.exists(tableId)) {
```

Comment:
You can remove the exists check. The will not exist

Reply:
Also, I think you can move the HelloWorld instantiation to beforeClass to avoid re-creating the table